### PR TITLE
[WIP] Core/Scripts: Thalorien Dawnseeker (q 24563/24563)

### PR DESCRIPTION
**Pull request by ghost**
Target branch(es): 335/6x

Issues addressed: closes issue #12444

Tests performed: (Does it build, tested in-game, etc) Builds OK, tested OK.

I express my gratitude for the help @tkrokli.
- [ ]  test for undetected bugs
- [ ] add improvements to the script
- [ ] verify script contents from sniffs

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp [hunk 1]
```diff
@@ -88,7 +90,351 @@ class npc_greengill_slave : public CreatureScript
...
+    NPC_QUEST_CREDIT            = 37601,
+    NPC_SUNWELL_DEFENDER        = 37211,
+    NPC_SCOURGE_ZOMBIE          = 37538,
+    NPC_GHOUL_INVADER           = 37539,
+    NPC_CRYPT_RAIDER            = 37541
+};
+
+enum Phase
+{
+    PHASE_SHIFT = 70193
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]: should be handled by `spell_area`

File: src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp [hunk 2]
```diff
@@ -88,7 +90,351 @@ class npc_greengill_slave : public CreatureScript
...
+class npc_thalorien_dawnseeker : public CreatureScript
+{
+    public:
+        npc_thalorien_dawnseeker() : CreatureScript("npc_thalorien_dawnseeker") { }
+
+        bool OnGossipHello(Player* player, Creature* creature) override
+        {
+            player->PrepareGossipMenu(creature, 0);
+
+            if ((player->GetQuestStatus(QUEST_THALORIEN_A) == QUEST_STATUS_INCOMPLETE) || (player->GetQuestStatus(QUEST_THALORIEN_H) == QUEST_STATUS_INCOMPLETE))
```
Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]: pleeeease use conditions

Comment (DevQuad) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]: How difficult to understand english) Here I need help.

File: src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp (hunk 1 above)
Comment (DevQuad) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]: Remove from the core phase and add spell_area base? Right?

Comment (joschiwald) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]: yes

File: src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp (hunk 2 above)
Comment (tkrokli) [inline src/server/scripts/EasternKingdoms/zone_isle_of_queldanas.cpp]:
```sql
-- conditions for gossip menu option 37552 / Thalorien Dawnseeker's Remains (entry 37552)
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`= 15 AND `SourceGroup`= 37552;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`,`SourceGroup`,`SourceEntry`,`SourceId`,`ElseGroup`,`ConditionTypeOrReference`,`ConditionTarget`,`ConditionValue1`,`ConditionValue2`,`ConditionValue3`,`NegativeCondition`,`ErrorType`,`ErrorTextId`,`ScriptName`,`Comment`) VALUES
(15,37552,0,0,0,28,0,24535,0,0,1,0,0,'npc_thalorien_dawnseeker','Show gossip option 37552 only if Quest 24535 (A) is NOT completed'),
(15,37552,0,0,0,28,0,24563,0,0,1,0,0,'npc_thalorien_dawnseeker','Show gossip option 37552 only if Quest 24563 (H) is NOT completed');
```

**Comment by joschiwald** [comment]
maybe that can help you (from sniff)
```sql
UPDATE `creature_template` SET `faction`=534 WHERE `entry`=37765;
UPDATE `creature_template` SET `gossip_menu_id`=10908 WHERE `entry`=37552; -- Thalorien Dawnseekers Remains
UPDATE `creature_template` SET `gossip_menu_id`=10963 WHERE `entry`=37523; -- Warden of the Sunwell

DELETE FROM `gossip_menu` WHERE `entry` IN (10908, 10963);
INSERT INTO `gossip_menu` (`entry`, `text_id`) VALUES
(10908, 15155), -- Thalorien Dawnseekers Remains
(10963, 15239), -- Warden of the Sunwell
(10963, 15240); -- Warden of the Sunwell

DELETE FROM `gossip_menu_option` WHERE `menu_id` IN (10908, 10963);
INSERT INTO `gossip_menu_option` (`menu_id`, `id`, `option_icon`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`,`action_menu_id`, `action_poi_id`, `box_coded`, `box_money`, `box_text`, `BoxBroadcastTextID`) VALUES
(10908, 0, 0, 'Examine the remains.', 37149, 1, 1, 0, 0, 0, 0, '', 0), -- 37552
(10963, 0, 0, 'I''m ready to enter the Sunwell.', 37806, 1, 1, 0, 0, 0, 0, '', 0); -- 37523

UPDATE `npc_text` SET `text0_0`='These appear to be the remains of Thalorien Dawnseeker, the last wielder of Quel''Delar.', `BroadcastTextID0`=37148 WHERE `id`=15155; -- Thalorien Dawnseekers Remains

DELETE FROM `npc_text` WHERE `ID` IN (15239, 15240);
INSERT INTO `npc_text` (`ID`, `text0_0`, `text0_1`, `BroadcastTextID0`, `prob0`, `VerifiedBuild`) VALUES
(15239, 'Only those who have the ranger general''s approval may enter the Sunwell Plateau.', 'Only those who have the ranger general''s approval may enter the Sunwell Plateau.', 37803, 1, 12340),
(15240, 'I will escort you into the Sunwell when you''re ready.', 'I will escort you into the Sunwell when you''re ready.', 37804, 1, 12340);
```

**Comment by mayaren888** [comment]
@DevQuad    Your code will have a serious problem, when the player to complete the task, will enter another space, can not see any NPC

**Comment by DevQuad** [comment]
Yes I know, this is due precisely to the phase. Once the add phase spell_area should work fine.

**Comment by mayaren888** [comment]
@DevQuad   Hope to repair as soon as possible, thank you.

**Comment by w5860363** [comment]
bool OnGossipSelect(Player* player, Creature* creature, uint32 /*uiSender*/, uint32 action) override
    {
        player->PlayerTalkClass->ClearMenus();

        switch (action)
        {
        case GOSSIP_ACTION_INFO_DEF + 1:
            player->CLOSE_GOSSIP_MENU();
            creature->AI()->SetGUID(player->GetGUID());
            creature->SetPhaseMask(2, true);
            player->SetPhaseMask(2, true); // Better handling if we find the correct Phasing-Spell
            creature->AI()->DoAction(ACTION_START_QUEST);
            break;
        default:
            return false;
        }
        return true;
    }

Phase can be set up temporary？

**Comment by DevQuad** [comment]
I finish my work at the branch 3.3.5, if someone can modify and want to pull a solution.

File: sql/updates/world/2016_03_01_NPC_THALORIEN_DAWNSEEKER_world.sql [hunk 3]
```diff
@@ -0,0 +1,79 @@
+-- Thalorien Dawnseeker's Remains
+SET @ENTRY := 37552;
+
+DELETE FROM `creature_template_addon` WHERE (`entry`=@ENTRY);
+INSERT INTO `creature_template_addon` (`entry`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`auras`) VALUES 
+(@ENTRY, 0, 0, 0, 0, 0, '25824 31261');
```
Comment (Expecto) [inline sql/updates/world/2016_03_01_NPC_THALORIEN_DAWNSEEKER_world.sql]: 25824 wrong spawn aura

look screen http://www.millenium.org/upload/queldelar16.jpg

Comment (Aokromes) [inline sql/updates/world/2016_03_01_NPC_THALORIEN_DAWNSEEKER_world.sql]: congrats you have posted on the PR of one user who deleted the account and a work never merged.

**Pull request closed without merging**